cmd/erasure-server-pool.go

```diff
@@ -1035,10 +1035,20 @@ func (z *erasureServerPools) NewMultipartUpload(ctx context.Context, bucket, obj
 		return z.serverPools[0].NewMultipartUpload(ctx, bucket, object, opts)
 	}
 
-	// We don't know the exact size, so we ask for at least 1GiB file.
-	idx, err := z.getPoolIdx(ctx, bucket, object, 1<<30)
-	if err != nil {
-		return "", err
+	for idx, pool := range z.serverPools {
+		result, err := pool.ListMultipartUploads(ctx, bucket, object, "", "", "", maxUploadsList)
```

Comment:
Nice!

A comment is really necessary here:
// If there is a multipart upload with the same bucket/object name, create the new multipart in the same pool, this will avoid creating two multiparts uploads in two different pools